### DISCUSSION TITLE
Why is my total count showing `1` when it should be `0`?

### DISCUSSION THREAD

**Discussion by aljawaid** [aljawaid/TemplateManager#6] [Q&A] [answered]
Everything else in total counts works, but if there are zero templates, it always shows as `1`

I tried...
```php
<?php
$taskTemplatesCount =  $this->task->taskDescriptionTemplateModel->getAll($project['id']);
$taskTemplatesCountTotal = count($taskTemplatesCount);
$commentTemplatesCount = $this->task->taskCommentTemplateModel->getAll($project['id']);
$commentTemplatesCountTotal = count($commentTemplatesCount);
$emailSubjectsCount = count(explode("\r\n", trim($project['predefined_email_subjects'])));
$allTemplatesCount = array($taskTemplatesCountTotal, $commentTemplatesCountTotal, $emailSubjectsCount);
?>
```

I also tried...
```php
<?php
$taskTemplatesCount =  $this->task->taskDescriptionTemplateModel->getAll($project['id']);
$taskTemplatesCountTotal = count($taskTemplatesCount);
$commentTemplatesCount = $this->task->taskCommentTemplateModel->getAll($project['id']);
$commentTemplatesCountTotal = count($commentTemplatesCount);
$emailSubjectsCount = count(explode("\r\n", trim($project['predefined_email_subjects'])));
$allTemplatesCount = ($taskTemplatesCountTotal + $commentTemplatesCountTotal + $emailSubjectsCount);
?>
```

https://github.com/aljawaid/TemplateManager/blob/c07e750cbea661ee55946d795213dff8a80b3f7b/Template/project_predefined_content/show.php#L23

**Comment by aljawaid** [comment]
@creecros could you kindly check the code?

**Comment by creecros** [comment, marked as answer]
2 things:
## 1st:
`$emailSubjectsCount = count(explode("\r\n", trim($project['predefined_email_subjects'])));`
is not PHP 8 compliant.

## 2nd:
`$emailSubjectsCount = count(explode("\r\n", trim($project['predefined_email_subjects'])));` will always equal at least 1

## Change to:
`$emailSubjectsCount = (!empty($project['predefined_email_subjects'])) ? count(explode("\r\n", trim($project['predefined_email_subjects']))) : 0;`

should fix both.

**Reply by creecros** [reply]
if you want to make sure it fixes both, maybe add `&& !is_null($project['predefined_email_subjects'])` to that.

**Reply by creecros** [reply]
i.e.

`$emailSubjectsCount = (!empty($project['predefined_email_subjects']) && !is_null($project['predefined_email_subjects'])) ? count(explode("\r\n", trim($project['predefined_email_subjects']))) : 0;`

**Reply by aljawaid** [reply]
```php
$emailSubjectsCount  = (!empty($project['predefined_email_subjects'])) ? count(explode("\r\n", trim($project['predefined_email_subjects']))) : 0;
```
Seemed to work without any errors.... big thanks for that

**Reply by creecros** [reply]
I would actually add the `&& !is_null($project['predefined_email_subjects'])` , and in your dropdown. It has PHP 8 errors popping up for me without it, on initial plugin use.

once you enter data into the set, and remove it, `!empty` is enough, but initially, it is null.

**Reply by aljawaid** [reply]
ok I will do it now

**Reply by aljawaid** [reply]
done, big thanks 563a0e40e1273e6d6f7e567da8b4d56449583da9
